### PR TITLE
Stratis.Bitcoin.Features.IndexStore and Tests into separate projects

### PR DESCRIPTION
**Pull request by quantumagi**
Two new projects have been created:
- Stratis.Bitcoin.Features.IndexStore (code from Stratis.Bitcoin project - same namespace)
- Stratis.Bitcoin.Features.IndexStore.Tests (tests from Stratis.Bitcoin.Tests.IndexStore)

### CODE REVIEW

File: Stratis.Bitcoin.Features.IndexStore/IndexStoreRPCController.cs [hunk 1]
```diff
@@ -16,8 +17,9 @@ public class IndexStoreRPCController : BaseRPCController
 
         public IndexStoreRPCController(
             ILoggerFactory loggerFactory,
-            IndexStoreManager indexManager = null)
-            : base(indexManager: indexManager)
+            IndexStoreManager indexManager,
+            MempoolManager mempoolManager = null)
```
Comment (bokobza) [inline Stratis.Bitcoin.Features.IndexStore/IndexStoreRPCController.cs]: do you need this MempoolManager here?

Comment (quantumagi) [inline Stratis.Bitcoin.Features.IndexStore/IndexStoreRPCController.cs]: No I probably don't need it at all. Will remove it.

File: Stratis.Bitcoin.Features.IndexStore/Stratis.Bitcoin.Features.IndexStore.csproj [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  <ItemGroup>
+    <ProjectReference Include="..\NStratis\NBitcoin\NBitcoin.csproj" />
+    <ProjectReference Include="..\Stratis.Bitcoin\Stratis.Bitcoin.csproj" />
+  </ItemGroup>
+
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard1.6' ">
+    <DefineConstants>$(DefineConstants);NETCORE</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net462' ">
+    <DefineConstants>$(DefineConstants);NOASSEMBLYCONTEXT</DefineConstants>
```
Comment (dangershony) [inline Stratis.Bitcoin.Features.IndexStore/Stratis.Bitcoin.Features.IndexStore.csproj]: You may be able to remove that

**Comment by bokobza** [comment]
@dangershony any reason why you committed this before it was changed?

**Pull request merged**